grpc-protocol/src/main/java/com/linecorp/armeria/client/grpc/protocol/UnaryGrpcClient.java

```diff
@@ -164,110 +167,163 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) {
-                                       throw new ArmeriaStatusException(StatusCodes.INTERNAL,
-                                                                        "Error executing request.");
+                                       try {
+                                           return unwrap().execute(ctx, HttpRequest.of(req.headers(), framed))
+                                                          .aggregateWithPooledObjects(ctx.eventLoop(),
+                                                                                      ctx.alloc());
+                                       } catch (Exception e) {
+                                           throw new ArmeriaStatusException(StatusCodes.INTERNAL,
+                                                                            "Error executing request.");
+                                       }
                                    }
                                })
                        .thenCompose(msg -> {
-                           if (!msg.status().equals(HttpStatus.OK) || msg.content().isEmpty()) {
-                               // Nothing to deframe.
-                               return CompletableFuture.completedFuture(msg.toHttpResponse());
-                           }
-
-                           final CompletableFuture<HttpResponse> responseFuture = new CompletableFuture<>();
+                           try (HttpData content = msg.content()) {
+                               if (msg.status() != HttpStatus.OK || content.isEmpty()) {
+                                   // Nothing to deframe.
+                                   return CompletableFuture.completedFuture(msg.toHttpResponse());
+                               }
 
-                           final ArmeriaMessageDeframer deframer =
-                                   new ArmeriaMessageDeframer(Integer.MAX_VALUE);
-                           msg.toHttpResponse().decode(deframer, ctx.alloc(),
-                                                       byteBufConverter(ctx.alloc(), isGrpcWebText))
-                              .subscribe(singleSubscriber(ctx, msg, serializationFormat, responseFuture),
-                                         ctx.eventLoop());
-                           return responseFuture;
+                               final CompletableFuture<HttpResponse> responseFuture = new CompletableFuture<>();
+                               final ArmeriaMessageDeframer deframer =
+                                       new ArmeriaMessageDeframer(Integer.MAX_VALUE);
+                               msg.toHttpResponse()
+                                  .decode(deframer, ctx.alloc(), byteBufConverter(ctx.alloc(), isGrpcWebText))
+                                  .subscribe(new DeframedMessageSubscriber(
+                                                     msg, serializationFormat, responseFuture),
+                                             ctx.eventLoop());
```

Comment:
This is not your fault but how about specifying `SubscriptionOption.WITH_POOLED_OBJECTS` when subscribing so that we can reduce the number of byte buf copies?

Reply:
Let me merge once this comment is addressed.

Reply:
Done.